paasta_tools/cli/cmds/status.py

```diff
@@ -1325,35 +1377,35 @@ def get_replica_state(pod: KubernetesPodV2) -> ReplicaState:
         if not pod.containers or all([c.state == "waiting" for c in pod.containers]):
             state = ReplicaState.STARTING
         else:
-            state = ReplicaState.UNHEALTHY
+            state = ReplicaState.NOT_RUNNING
     elif phase == "Running":
-        if not pod.ready:
-            # TODO: Take sidecar containers into account
-            #   This logic likely needs refining
-            state = ReplicaState.UNREACHABLE
-            main_container = [
-                c
-                for c in pod.containers
-                if c.name not in kubernetes_tools.SIDECAR_CONTAINER_NAMES
-            ]
-            if main_container:
-                if main_container[0].restart_count > 0:
-                    if (
-                        pod.create_timestamp
-                        + main_container[0].healthcheck_grace_period
-                        > datetime.utcnow().timestamp()
-                    ):
-                        state = ReplicaState.WARMING_UP
-                    else:
-                        state = ReplicaState.HEALTHCHECK_FAILING
-                else:
-                    # No restarts yet, still warming up?
-                    state = ReplicaState.WARMING_UP
+        ####
+        #
+        # TODO: Take sidecar containers into account
+        #   This logic likely needs refining
+        main_container = get_main_container(pod)
+        if main_container:
+            warming_up = (
+                pod.create_timestamp + main_container.healthcheck_grace_period
+                > datetime.now().timestamp()
+            )
+            if main_container.state != "running" or not pod.ready:
+                state = ReplicaState.NOT_RUNNING
+            elif pod.mesh_ready is False:  # Not registered if None
+                state = ReplicaState.UNREACHABLE
+            elif recent_liveness_failure(pod) or recent_container_restart(
+                main_container
+            ):
+                state = ReplicaState.WARNING
+            elif warming_up:
+                state = ReplicaState.WARMING_UP
```

Comment:
I think that if its within the grace period and its not in the mesh, we should call it warming up -- during the warmup period it is allowed to fail its healthcheck, which will result in it not being registered in the mesh.

Reply:
Done -- in order to make sure we still show Unreachable rather than WARNING when a pod not in mesh but failing liveness probe (possibly not enough to cause a container restart), I left the ordering here but validated we aren't `warming_up`.